### PR TITLE
fix: Combine and rate limit invalid payload erros to sentry

### PR DESCRIPTION
**Pull request by tiina303**
## Problem

We're getting spammed occasionally by bad payloads, there's two problems this pr tries to fix:
1. there's many different issues in sentry
2. spamming sentry when we get spammed - we should rate limit

related thread: https://posthog.slack.com/archives/C0185UNBSJZ/p1654804890602569

Second example of bad payloads is bad `sent_at` value, again annoyingly here we created lots of different sentry errors, we'll want to combine and rate limit those as well (and ignoring the bad value here is fine) e.g. 
1. https://sentry.io/organizations/posthog2/issues/3305077666/?project=1899813&referrer=slack
2. https://sentry.io/organizations/posthog2/issues/3223408228/?query=is%3Aunresolved&sort=freq&statsPeriod=1h
3. https://sentry.io/organizations/posthog2/issues/3228754841/?project=1899813&query=is%3Aunresolved+valueerror&statsPeriod=14d

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #
-# This file is autogenerated by pip-compile with python 3.8
+# This file is autogenerated by pip-compile with python 3.9
```
Comment (fuziontech) [inline requirements.txt:2]: def make sure you are using python 3.8 locally to match prod

File: posthog/api/utils.py [hunk 2]
```diff
@@ -126,12 +127,20 @@ def get_project_id(data, request) -> Optional[int]:
     return None
 
 
+@limits(calls=10, period=1)
+def capture_as_common_error_to_sentry(error):
+    try:
+        raise Exception("Invalid payload") from error
```
Comment (fuziontech) [inline posthog/api/utils.py:133]: Definitely pass along the `error` here so that we have some context from the sample

Comment (tiina303) [reply posthog/api/utils.py:133]: if we raise this from error, then we do get that in the stacktrace, no?

Comment (fuziontech) [reply posthog/api/utils.py:133]: oh nm, this is a great exception chain! IGNORE ME

**Review by fuziontech — APPROVED** [review]
lgtm!

**Comment by macobo** [comment]
To combine errors in sentry, code level changes are the wrong way to go. Instead we should be using the fingerprinting feature.

**Pull request merged**